ink-wrapper-types/src/calls.rs

```diff
@@ -150,13 +153,33 @@ pub struct ReadCall<T: scale::Decode + Send> {
 
 impl<T: scale::Decode + Send> ReadCall<T> {
     /// Create a new read call.
+    pub fn new(account_id: AccountId, data: Vec<u8>) -> Self {
+        Self {
+            account_id,
+            data,
+            value: 0,
+            _return_type: Default::default(),
+        }
+    }
+}
+
+impl<T: scale::Decode + Send> ReadCallNeedsValue<T> {
+    /// Create a new needs value call.
     pub fn new(account_id: AccountId, data: Vec<u8>) -> Self {
         Self {
             account_id,
             data,
             _return_type: Default::default(),
         }
     }
+
+    /// Set the value to be sent with the call.
+    pub fn with_value(self, value: u128) -> ReadCall<T> {
+        ReadCall {
+            value,
+            ..ReadCall::new(self.account_id, self.data)
+        }
+    }
```

Comment:
Suggestion: remove. As stated in priv, I don't think it makes sense. It is theoretically possible to have a call that does not mutate the state, but depends on value passed anyway. However, I would contend we should discourage such a thing too. I'd propose not doing it until someone comes along with an actual usecase.

Reply:
Done [Remove value from ReadCall](https://github.com/Cardinal-Cryptography/ink-wrapper/pull/68/commits/169e0d57bcbea3b531d2a219302597d26879ee78)